backends/core/workers.py

```diff
@@ -1842,3 +1843,65 @@ def _set_target_column(self, client, dataset_resource_name, target_column_spec):
     response = client.projects().locations().datasets() \
                      .patch(name=dataset_resource_name, body=body).execute()
     self.log_info('Modified target column for dataset: %s', response)
+
+class AdsDataHubWorker(Worker):
+  """Abstract AdsDataHub worker."""
+
+  GLOBAL_SETTINGS = ['developer_token']
+  ADH_SCOPE = 'https://www.googleapis.com/auth/adsdatahub'
+
+  def _get_adh_client(self):
+    credentials = ServiceAccountCredentials.from_json_keyfile_name(_KEY_FILE,
+                                                                   scopes=[ADH_SCOPE])
+    self._adh_client = build('adsdatahub',
+                             'v1',
+                             credentials=credentials,
+                             developerKey=self._params['developer_token'].strip())
+
+class AdsDataHubQueryLauncher(AdsDataHubWorker):
+  """AdsDataHub worker that launches a pre-existing query."""
+
+  PARAMS = [
+    ('customer_id', 'string', True, '', 'Customer Id'),
+    ('query_id', 'string', True, '', 'Query Id'),
+    ('payload', 'text', True, '', 'Payload')
+  ]
+
+  def _execute(self):
+    self._get_adh_client()
+
+    operation = self._adh_client.customers().analysisQueries().start(
+        name='customers/{customer_id}/analysisQueries/{query_id}'
+          .format(
+              query_id=self._params['query_id'],
+              customer_id=self._params['customer_id']),
+          body=json.loads(self._params['payload'])
+    ).execute()
+
+    self.log_info("New operation %s created", operation['name'])
+
+    self._enqueue('AdsDataHubQueryWaiter',
+                    {'operation_name': operation['name']}, 60)
+
+class AdsDataHubQueryWaiter(AdsDataHubWorker):
+  """Worker that keeps respawning until an AdsDataHub operation is done."""
+
+  PARAMS = [
+    ('operation_name', 'string', True, '', 'Operation name'),
+  ]
+
+  def _execute(self):
+    self._get_adh_client()
+
+    operation = self._adh_client.operations().get(
+        name=self._params['operation_name']).execute()
+
+    # API is not behaving as expected, i.e. done is not added to response
+    # before the execution is finished
+    if 'done' not in operation or operation['done'] != True:
```

Comment:
You could simplify that in python like this

```python
if not operation.get('done', False):
  ...
```

With `False` being the default value used if 'done' is not in operation.